### PR TITLE
Adding support to white list or black black list certain twitch usernames

### PR DESCRIPTION
**Pull request by nckackerman**
Testing:
- logged on with a whitelisted account to a whitelisted env. Confirmed login worked as expected.
- logged in without whitelisted account to a whitelisted env. Confirmed I didnt login and was routed to the /login page
- logged in with a blacklisted account. Confirmed I didnt login and was routed to the /login page
- logged in without a blacklisted account. Confirmed login worked as expected.

### CODE REVIEW

File: api/src/main/java/com/nicknathanjustin/streamercontracts/login/LoginApiController.java [hunk 1]
```diff
@@ -33,7 +35,9 @@ public void ping(@NonNull final HttpServletRequest request,
         // Since its difficult to move the redirect, we'll have "/" function as the OauthSuccessRedirect when we see
         // a request from our Oauth login flow
         if(isOauthSuccessRedirect(auth, request)) {
-            loginService.logUserIn(auth);
+            if (!securityService.isAnonymousRequest(request)) {
```
Comment (m1sterc001guy) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/login/LoginApiController.java]: Why was this added and what is it doing?

File: api/src/main/java/com/nicknathanjustin/streamercontracts/security/SecurityServiceImpl.java [hunk 2]
```diff
@@ -1,21 +1,66 @@
...
+    private boolean isRecognizedUserBlocked(@NonNull final HttpServletRequest httpServletRequest) {
+        final UserModel userModel = userService.getUserModelFromRequest(httpServletRequest);
+        if (isEnvLockedToWhiteListedAccounts()) {
+            return !isUserWhiteListed(userModel);
+        }
+        return isUserBlackListed(userModel);
+    }
+
+    private boolean isEnvLockedToWhiteListedAccounts() {
+        // Beta endpoint is currently open to any client. However, we want to only allow bountyStreamer accounts to
```
Comment (m1sterc001guy) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/security/SecurityServiceImpl.java]: Whitelist is for beta, blacklist is for prod?

File: api/src/main/resources/application-beta.yml [hunk 3]
```diff
@@ -13,4 +13,7 @@ spring:
 
 paypal:
   clientId: AVbr-mNnSiaEKeAa97MPVerIi_G18U8GMqq8XyjXvAfRKH2bLhMy_Ue3XOtdNyjWScSalu8JoN3YJrnP
-  mode: sandbox
\ No newline at end of file
+  mode: sandbox
+
+twitch:
+  whiteListedAccounts: nckackerman, M1sterc001guy, vgame18
```
Comment (m1sterc001guy) [inline api/src/main/resources/application-beta.yml]: We might not want to check this into source control. Would it be difficult to keep the whitelist and blacklist on S3 so only we can see it?

File: api/src/main/java/com/nicknathanjustin/streamercontracts/login/LoginApiController.java (hunk 1 above)
Comment (nckackerman) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/login/LoginApiController.java]: When someone sign up for our site the flow goes. frontend -> twitch -> our backend (and specifically this method)

So if this check isnt there, a blacklisted user would still be able to hit the loginService.logUserIn(auth); method and have their account added to our database

File: api/src/main/java/com/nicknathanjustin/streamercontracts/security/SecurityServiceImpl.java (hunk 2 above)
Comment (nckackerman) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/security/SecurityServiceImpl.java]: yiss

File: api/src/main/resources/application-beta.yml (hunk 3 above)
Comment (nckackerman) [reply api/src/main/resources/application-beta.yml]: Good callout! This doesnt need to be source controlled and shouldnt be. Ive moved this list to the S3 secrets bucket.
